Looport/Millennium-Falcon
# Add registration logic

# Description
This pull request includes login logic. The user could send credentials to the `/authentication/register` route and JWT Token would be returned. When email already exists in the system, a Validation error will be thrown.

As part of this task, was implemented switching to native node test runner with using the `swc` compiler instead `tsc`. This change provides to run node test runner with using `typescript` in a simple way and improves compilation speed.

## What Changed
— Switched from `tsc` to the `swc`
— Added Login logic
— Added   test cases for the Login logic
— Added custom `ValidationPipe` for handling `class-validator` exceptions and transform errors
— Added `ValidationExeption` for handling custom `Validation` error

## How To Test
1. Send `email` and `password` to the `/authentication/login` route. The token should be returned.
2. When `email` already exists, then validation error should be throws with status `400`.

# Checklist
- [x] I followed _user flow_ of implemented code
